src/sentry/dynamic_sampling/sliding_window_rebalancing.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import List, Mapping, Optional, Sequence, Tuple
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    LimitBy,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry import options
+from sentry.dynamic_sampling.rules.utils import OrganizationId, ProjectId
+from sentry.dynamic_sampling.snuba_utils import MAX_TRANSACTIONS_PER_PROJECT
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 9998  # Snuba's limit is 10000, and we fetch CHUNK_SIZE+1
+QUERY_TIME_INTERVAL = timedelta(
+    hours=24
+)  # By default, we want to get the volume of the last 24 hours.
+
+
+def fetch_projects_with_total_root_transactions_count(
+    org_ids: List[int],
+    granularity: Optional[Granularity] = None,
+    query_interval: Optional[timedelta] = None,
+) -> Mapping[OrganizationId, Sequence[Tuple[ProjectId, int]]]:
+    """
+    Fetches tuples of (org_id, project_id) and the respective root transaction counts.
+    """
+    if query_interval is None:
+        query_interval = QUERY_TIME_INTERVAL
+        granularity = Granularity(3600)
+
+    count_per_root_metric_id = indexer.resolve_shared_org(
+        str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value)
+    )
+    where = [
+        Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - query_interval),
+        Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+        Condition(Column("metric_id"), Op.EQ, count_per_root_metric_id),
+        Condition(Column("org_id"), Op.IN, list(org_ids)),
+    ]
+
+    # Since we want a uniform distribution of samples, we use the modulo operator and sample with a given sample rate
+    # the orgs.
+    sample_rate = int(options.get("dynamic-sampling.sliding_window_rebalancing.sample_rate") * 100)
+    if sample_rate != 100:
+        where += [Condition(Function("modulo", [Column("org_id"), 100]), Op.LT, sample_rate)]
```

Comment:
I think we can safely remove this.
It was initially done because we were not sure the DB will be able to cope with this query.

Since we are doing basically the same query in Prioritise Projects and we do a lot more for transactions we know there are no DB projects so there is no need (IMO) to bother with this.

Reply:
Will do. I have put it in place since maybe we wanted a gradual rollout but since this query is even less expensive than the one before, I don't expect any regressions.